### ISSUE TITLE
After upgrade to NC21: BackgroundJob fails wrong argument to `CirclesRequest->getCircle()`

### ISSUE THREAD

**Issue by doobry-systemli** [nextcloud/deck#2955]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**
After upgrading from Nextcloud 20.0.8 to 21.0.1, a lot of Exceptions get logged by the deck background job `ScheduleNotifications`:

```yaml
{
  "Exception": "TypeError",
  "Message": "Argument 2 passed to OCA\\Circles\\Db\\CirclesRequest0:0:0:0:0:0:0:0getCircle() must be of the type string, null given, called in /var/www/cloud.example.org/nextcloud-21.0.1/apps/circles/lib/Service/CirclesService.php on line 277",
  "Code": 0,
  "Trace": [
    {
      "file": "/var/www/cloud.example.org/nextcloud-21.0.1/apps/circles/lib/Service/CirclesService.php",
      "line": 277,
      "function": "getCircle",
      "class": "OCA\\Circles\\Db\\CirclesRequest",
      "type": "->",
      "args": [
        "4346fee3223491",
        null,
        1,
        "",
        true
      ]
    },
    {
      "file": "/var/www/cloud.example.org/nextcloud-21.0.1/apps/circles/lib/Api/v1/Circles.php",
      "line": 265,
      "function": "detailsCircle",
      "class": "OCA\\Circles\\Service\\CirclesService",
      "type": "->",
      "args": [
        "4346fee3223491",
        true
      ]
    },
    {
      "file": "/var/www/cloud.example.org/nextcloud-21.0.1/apps/deck/lib/Service/PermissionService.php",
      "line": 265,
      "function": "detailsCircle",
      "class": "OCA\\Circles\\Api\\v1\\Circles",
      "type": "0:0:0:0:0:0:0:0",
      "args": [
        "4346fee3223491",
        true
      ]
    },
    {
      "file": "/var/www/cloud.example.org/nextcloud-21.0.1/apps/deck/lib/Notification/NotificationHelper.php",
      "line": 98,
      "function": "findUsers",
      "class": "OCA\\Deck\\Service\\PermissionService",
      "type": "->",
      "args": [
        78
      ]
    },
    {
      "file": "/var/www/cloud.example.org/nextcloud-21.0.1/apps/deck/lib/Cron/ScheduledNotifications.php",
      "line": 62,
      "function": "sendCardDuedate",
      "class": "OCA\\Deck\\Notification\\NotificationHelper",
      "type": "->",
      "args": [
        {
          "id": 2376,
          "__class__": "OCA\\Deck\\Db\\Card"
        }
      ]
    },
    {
      "file": "/var/www/cloud.example.org/nextcloud-21.0.1/lib/private/BackgroundJob/Job.php",
      "line": 52,
      "function": "run",
      "class": "OCA\\Deck\\Cron\\ScheduledNotifications",
      "type": "->",
      "args": [
        null
      ]
    },
    {
      "file": "/var/www/cloud.example.org/nextcloud-21.0.1/cron.php",
      "line": 128,
      "function": "execute",
      "class": "OC\\BackgroundJob\\Job",
      "type": "->",
      "args": [
        {
          "__class__": "OC\\BackgroundJob\\JobList"
        },
        {
          "__class__": "OC\\Log"
        }
      ]
    }
  ],
  "File": "/var/www/cloud.example.org/nextcloud-21.0.1/apps/circles/lib/Db/CirclesRequest.php",
  "Line": 192,
  "CustomMessage": "Error while running background job (class: OCA\\Deck\\Cron\\ScheduledNotifications, arguments: )"
}
```

It seems like argument 2 passed to `CirclesService->getCircle()` should be `$userId` but instead is empty. I'm not sure whether this is a bug in deck or in circles, but at least it seems to break deck notifications if a deck is shared with a circle.

<details>
<summary>Server details</summary>

**Operating system**: Debian Buster

**Web server:** Apache2

**Database:** MariaDB

**PHP version:** 7.3.27-1~deb10u1

**Nextcloud version:** 21.0.1

**Where did you install Nextcloud from:** Release tarball

**List of activated apps:**

```
  - circles: 0.21.0
  - dav: 1.17.1
  - deck: 1.3.2
```

**Are you using an external user-backend, if yes which one:** OC_User_IMAP

</details>

**Comment by doobry-systemli** [comment]
This is a bug in the Circles app that got reported as nextcloud/circles#563 and fixed in PR nextcloud/circles#566 already. But unfortunately there's no new Circles app release that incorporates this fix yet, so the issue remains unfixed on production systems.

**Comment by juliushaertl** [comment]
Yes, the release is still pending for the circles app so I'll close this one as it is not an issue in deck itself.
